perllib/FixMyStreet/Roles/CobrandSLWP.pm

```diff
@@ -803,9 +802,30 @@ sub waste_display_payment_method {
     return $display->{$method};
 }
 
+sub _get_cost_from_array {
+    my ($self, $costs) = @_;
+
+    my $now = DateTime->now->set_time_zone(FixMyStreet->local_time_zone);
+    my @sorted = sort { $b->{start_date} cmp $a->{start_date} } @$costs;
+    foreach my $cost (@sorted) {
+        my $start_date = DateTime::Format::Strptime->new(
```

Comment:
Also to deal with your "need to provide a date in the past" comment, not that it matters either, but instead of the die you could return the last entry in @sorted assuming you'd want the earliest date.

Reply:
Tried doing this but still needed a `start_date` for the `sort` comparison, so leaving this as-is for now in the interests of getting this live today 🙂